# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Lane:** CTR / Engagement Opportunity Scoring

Pages that rank well but get fewer clicks than expected are a clear problem worth fixing. 
This lane gives a ranked list of pages to work on first, so it is easy to turn into action. 
I am picking this lane because the decision, the action, and the cost of a wrong call are 
all easy to explain.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

For the SEO team, deciding which pages to improve first, we will build a ranked priority 
score from position, impressions, and CTR data, scoring each page by how much opportunity 
it has, measured by how well the top-ranked pages match real problem pages (precision@K). 
A wrong call costs wasted time fixing a page that was already fine, or missing a page that 
was quietly losing clicks. A simple rule is not enough because "underperforming" depends on 
position and volume together — one fixed CTR cutoff would wrongly flag high-position pages 
and miss real problems lower down.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
import pandas as pd

df = pd.read_csv("C:/Users/tayyaba/Desktop/Flyrank-ML-Internship/data/raw/content_refresh_anonymized.csv")

# avg_position = 0 means "no data", so remove those rows
has_position = df[df["avg_position"] > 0]

# look only at pages ranking in the top 10
top10 = has_position[has_position["avg_position"] <= 10]

# find the middle CTR value for top 10 pages
median_ctr = top10["ctr"].median()

# pages that rank well but have below-average clicks
low_ctr_pages = top10[top10["ctr"] < median_ctr]

print("Total pages:", len(df))
print("Pages ranking top 10:", len(top10))
print("Of those, below-median CTR:", len(low_ctr_pages))
print("Percent underperforming:", round(len(low_ctr_pages) / len(top10) * 100, 1), "%")

Total pages: 30000
Pages ranking top 10: 12983
Of those, below-median CTR: 6432
Percent underperforming: 49.5 %


Out of 30,000 pages, 12,983 rank in the top 10 positions. Of those, 6,432 pages 
(49.5%) get fewer clicks than the median, even though they rank well. This shows 
a real gap worth fixing — good ranking position does not always mean good clicks, 
and there are enough cases here to make this lane worth building on.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

A simple rule like "flag anything under 2% CTR" would not work well, because expected 
CTR changes a lot with position — a page at position 3 and a page at position 9 have 
very different normal CTR ranges. A scoring model can account for position and volume 
together, instead of using one fixed number for every page.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.